### PR TITLE
Jakarta Tags 3.0 Spec

### PR DESCRIPTION
**Pull request by volosied**
## Specification PR template
When creating a specification project release review, create PRs with the content defined as follows.

Include the following in the PR:
- [x] A directory in the form wombat/x.y where x.y is the release major.minor version, and the directory contains the following.
- [x] Specification PDF in the form of jakarta-wombat-spec-x.y.pdf
- [x] Specification HTML in the form of jakarta-wombat-spec-x.y.html
- [x] A specification page named _index.md following the template at:
      https://github.com/jakartaee/specification-committee/blob/master/spec_page_template.md
- [ ] For a Progress Review, that sufficient progress has been made on a Compatible Implementation and TCK, to ensure that the spec is implementable and testable.
- [x] For a [Release Review](https://www.eclipse.org/projects/handbook/#release-review), a summary that a Compatible Implementation is complete, passes the TCK, and that the TCK includes sufficient coverage of the specification. The TCK users guide MUST include the instructions to run the compatible implementations used to validate the release.
Instructions MAY be by reference.
   - [x] Updated release record
   - [x] Generated IP Log
   - [x] Email to PMC
   - [ ] Start release review by emailing EMO 
- [x] The URL of the OSSRH staging repository for the api, javadoc:
      https://jakarta.oss.sonatype.org/content/repositories/staging/jakarta/servlet/jsp/jstl/jakarta.servlet.jsp.jstl-api/3.0.0/
- [x] The URL of the staging directory on downloads.eclipse.org for the proposed EFTL TCK binary:
      https://download.eclipse.org/ee4j/jakartaee-tck/jakartaee10/staged/eftl/jakarta-tags-tck-3.0.0.zip
- [x] The URL of the compatibility certification request issue:
      https://github.com/eclipse-ee4j/jstl-api/issues/226
- [x] Specification JavaDoc in the wombat/x.y/apidocs directory. 
If desired, an optional second PR can be created to contain just the JavaDoc in the `apidocs` directory.

Note: If any item does not apply, check it and mark N/A below it.

### CODE REVIEW

**Comment by volosied** [comment]
Reminder: https://github.com/eclipse-ee4j/jakartaee-tck/issues/882#issuecomment-1064478086

**Comment by starksm64** [comment]
# Spec Review Checklist (Mentor)

1. Spec PR
  - [x] PR uses [template](https://github.com/jakartaee/specifications/blob/master/pull_request_template.md)
  - [x] Directory of form {spec}/x.y
  - [x] PDF of form jakarta-{spec}-spec-x.y.pdf ("-spec" preferred but not required)
  - [x] HTML of form jakarta-{spec}-spec-x.y.html ("-spec" preferred but not required)
  - [ ] Index page {spec}/x.y/_index.md following [template](https://github.com/jakartaee/specification-committee/blob/master/spec_page_template.md)
  - [x] Index page {spec}/_index.md following [template](https://github.com/jakartaee/specification-committee/blob/master/spec_index_template.md)
  - [x] No other files (e.g., no jakarta_ee_logo_schooner_color_stacked_default.png)
  - [ ] Staging repository link of the form https://jakarta.oss.sonatype.org/content/repositories/staging/jakarta/{spec}/jakarta.{spec}-api/x.y.z/
  - [ ] EFTL TCK link of the form http://download.eclipse.org/.../+.zip
  - [ ] Compatibility certification link of the form https://github.com/eclipse-ee4j/{project}/#{issue}
  - [ ] (Optional) Second PR for just apidocs

2. _index.md
  - [ ] Link to project release plan of the form https://projects.eclipse.org/projects/ee4j.{spec}/releases/x.y/plan
  - [x] Link to spec pdf
  - [x] Link to spec html
  - [x] Link to apidocs
  - [ ] Link to final TCK download zip file of the form https://download.eclipse.org/jakartaee/{spec}/x.y/*{spec}-tck-x.y.z.zip (The folder path is required, the file name pattern is preferred.)
  - [ ] Link to API jar file of the form https://search.maven.org/artifact/jakarta.{spec}/jakarta.{spec}-api/x.y.z/jar
  - [ ] Name of and link to at least one Compatible Implementation
  - [ ] Review summary statement to ensure it has been updated to reflect the work accomplished and does not contain proposal statements as might be found in Plan Review statement.

3. javadocs
  - [ ] Footer contains Eclipse copyright and link to license
  - [ ] ESFL license is included, usually as doc-files/speclicense.html
  - [ ] no META-INF directory in PR
  - [ ] javadocs-jar artifact matches apidocs (optional for this release)

4. Spec PDF
  - [ ] Correct spec title
  - [ ] Version number of the form x.y, not x.y.z
  - [ ] Correct Eclipse copyright line
  - [ ] No DRAFT or SNAPSHOT
  - [ ] Correct Logo

5. Spec HTML
  - [ ] Same as PDF

6. TCK zip file
  - [ ] README file (optional for this release)
  - [ ] EFTL license file, preferably named LICENSE.md
  - [ ] User's Guide (or equivalent documentation)
  - [ ] How to test the Compatible Implementation(s) listed in _index.md above with the TCK (may be in UG)

7. TCK User's Guide (or equivalent documentation)
  - [ ] Software requirements listed
  - [ ] Installation and configuration described
  - [ ] How to run tests
  - [ ] Where to file challenges

8. Compatibility certification request
  - [ ] Request follows [template](https://github.com/jakartaee/specification-committee/blob/master/compatibility-certification-request.md)
  - [ ] SHA-256 fingerprint matches staged TCK zip file
  - [ ] Request issue has `certification` label.

9. TCK results summary
  - [ ] Page is hosted by Compatible Implementation project
  - [ ] Includes all information from certification request
  - [ ] Summary includes number of tests passed, failed, errors
  - [ ] SHA-256 fingerprint matches staged TCK zip file on cert request

10. If a Release Review is required, the specification project team contacts the EMO to initiate the review by sending an email to emo@eclipse.org.
    (A Release Review is not required if the current release is a Service Release based on a previously successful Major or Minor
    release as indicated by a release record on the project's Releases page, e.g., the [Jakarta Servlet releases page](https://projects.eclipse.org/projects/ee4j.servlet/reviews).)
    - [ ] An issue will be created by the EMO to track the release review.
    - [ ] The specification project team requests approval for the release from the PMC by sending an email to ee4j-pmc@eclipse.org.
    - [ ] The specification project team then delivers an IP Log to the IP Team for their review as described in https://www.eclipse.org/projects/handbook/#pmi-commands-iplog.

11. Update Jakarta EE API jar
  - [ ] Update the Jakarta EE API jar by submitting a PR to the jakartaee-api project that updates the version number of your API jar file.

**Comment by starksm64** [comment]
@volosied the specification pdf and html are using DRAFT as their status. Can you update these to either have no status line or Final?

**Comment by volosied** [comment]
@starksm64 Just updated them to final.

**Comment by volosied** [comment]
@starksm64  

Staging repository link of the form [https://jakarta.oss.sonatype.org/content/repositories/staging/jakarta/{spec}/jakarta.{spec}-api/x.y.z/](https://jakarta.oss.sonatype.org/content/repositories/staging/jakarta/%7Bspec%7D/jakarta.%7Bspec%7D-api/x.y.z/)

Here's the link: https://jakarta.oss.sonatype.org/content/repositories/staging/jakarta/servlet/jsp/jstl/jakarta.servlet.jsp.jstl-api/3.0.0/ 

Should this be check-marked?  or is something incorrect? 
____

The Jakarta Tags TCK is passing. How can I move forward with it?  (Zip, test results, etc) 

- TCK results summary
- TCK zip file
- EFTL TCK link of the form http://download.eclipse.org/.../+.zip
Compatibility certification link of the form [https://github.com/eclipse-ee4j/{project}/#{issue}](https://github.com/eclipse-ee4j/%7Bproject%7D/#%7Bissue%7D)

**Comment by volosied** [comment]
Scott posted this -- https://github.com/eclipse-ee4j/jakartaee-tck/pull/886#issuecomment-1104024503

**Comment by starksm64** [comment]
So now what is needed is a candidate TCK bundle and a CCR for a canidate GlassFish release. Let me ask @scottmarlow about that.

**Comment by scottmarlow** [comment]
> So now what is needed is a candidate TCK bundle 

Standalone TCK http://download.eclipse.org/ee4j/jakartaee-tck/jakartaee10/staged/eftl//jakarta-tags-tck-3.0.0.zip was updated today and the nightly GlassFish build (https://download.eclipse.org/ee4j/glassfish/glassfish-7.0.0-SNAPSHOT-nightly.zip) passed.

> and a CCR for a canidate GlassFish release. Let me ask @scottmarlow about that.

Test results are here but not in CCR form:  https://ci.eclipse.org/jakartaee-tck/job/10/job/eftl-standalonetck-build-run-100/106/artifact/results/junitreports/tags-tck-junit-report.xml/*view*/

From looking for previous CCRs via https://github.com/eclipse-ee4j/jstl-api/issues?q=is%3Aissue+is%3Aclosed+compatibility+label%3Acertification, I see that the last CCR https://github.com/eclipse-ee4j/jstl-api/issues/126 referred to TCK results copied to https://eclipse-ee4j.github.io/jstl-api/certifications/jakarta-tags/2.0/TCK-Results

**Comment by volosied** [comment]
@arjantijms @starksm64   What are the next steps for the CCR? Also, what about the other items in the checklist list? 
- Updated release record (can someone explain this more in detail?) 
- Generated IP Log (How do we do this?)  
- Email to PMC 
- Start release review by emailing EMO

Thanks!

**Comment by arjantijms** [comment]
You can generate the IP log via the Eclipse project page. There's a link for it on the right side of the page. The preview looks like this: https://projects.eclipse.org/projects/ee4j.jstl/releases/3.0.0/iplog/preview

IP log: https://dev.eclipse.org/ipzilla/show_bug.cgi?id=24062

**Comment by arjantijms** [comment]
PMC Mail: https://www.eclipse.org/lists/ee4j-pmc/msg03320.html

File: tags/3.0/_index.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 ---
-title: "Jakarta Standard Tag Library 3.0 (under development)"
-date: 2021-04-15
+title: "Jakarta Standard Tag Library 3.0"
+date: 2022-02-24
```
[the remark this answers is not in the document]
Comment (starksm64) [reply tags/3.0/_index.md:3]: Set it to the date of the final specification.

**Comment by starksm64** [comment]
This looks good to me.

File: tags/3.0/_index.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (volosied) [reply tags/3.0/_index.md]: Done. Thanks!

**Comment by volosied** [comment]
@starksm64 @arjantijms  Please let me know if I can do anything to help.  I'm not sure what else specifically needs to be done / followed up on. Thanks again for all your help!

**Comment by starksm64** [comment]
@volosied this looks ready to go to ballot to me, so I have asked @Emily-Jiang to start the ballot. That is the next step in the release process.

**Comment by arjantijms** [comment]
Release review tracking issues: https://gitlab.eclipse.org/eclipsefdn/emo-team/emo/-/issues/277

File: tags/3.0/_index.md [hunk 2]
```diff
@@ -9,7 +9,7 @@ for manipulating XML documents, internationalization tags, and SQL tags. It also
 existing custom tags with Jakarta Standard Tag Library tags.
 
 * [Jakarta Standard Tag Library 3.0 Release Record](https://projects.eclipse.org/projects/ee4j.jstl/releases/3.0.0)
```
Comment (volosied) [inline tags/3.0/_index.md:11]: @ivargrimstad Wanted to address your comments mentioned here: https://www.eclipse.org/lists/jakarta.ee-spec/msg02435.html

I should remove this `Release Record` above?

File: tags/3.0/_index.md [hunk 3]
```diff
@@ -22,22 +22,25 @@ existing custom tags with Jakarta Standard Tag Library tags.
...
 The plan for the release is to clean up open issues:
 * General specification document cleanup.
-* Clean up issues found during the Jakarta EE9 review ballot process: update if possible the tagdocs to include EFSL, Update tlddoc generation if possible to jakarta.
+* Clean up issues found during the Jakarta EE9 review ballot process
+* Update tagdocs to include EFSL
+* Update tlddoc generation to jakarta.
 * Remove any deprecated methods from the API if they exist.
 * Rename java.sun.com / xmlns.jcp.org taglib URIs to new jakarta.tags.\* URNs. [Issue 144](https://github.com/eclipse-ee4j/jstl-api/issues/144)
     * This alone should require a major release.
 * Other small clean up issues
```
Comment (volosied) [inline tags/3.0/_index.md:31]: @ivargrimstad 
Should I update this section to list a `Release Summary` (and include list of changes) or should I just remove entire section? 

Pages 3.1 and a few others, for example, don't include a Plan or other summary.

https://deploy-preview-435--jakartaee-specifications.netlify.app/specifications/pages/3.1/

File: tags/3.0/_index.md [hunk 4]
```diff
@@ -48,7 +51,7 @@ The following TCK updates will be required:
 * [Jakarta Standard Tag Library 3.0 Release Record](https://projects.eclipse.org/projects/ee4j.jstl/releases/3.0.0)
 
 # Compatible Implementations
-* [EE4J implementation of Jakarta Standard Tag Library 3.0.0](https://github.com/eclipse-ee4j/jstl-api/releases/download/3.0.0-IMPL-RELEASE/jakarta.servlet.jsp.jstl-3.0.0.jar) using vehicle [Eclipse Glassfish x.0.0-x](https://github.com/eclipse-ee4j/glassfish/releases/download/)
+* [EE4J implementation of Jakarta Standard Tag Library 3.0.0](https://github.com/eclipse-ee4j/jstl-api/releases/tag/3.0.0-IMPL-RELEASE)
```
Comment (volosied) [inline tags/3.0/_index.md:54]: FYI: Updated this to say our latest impl (as that's what was used against the TCK)

File: tags/3.0/_index.md [hunk 5]
```diff
@@ -18,17 +17,17 @@ existing custom tags with Jakarta Standard Tag Library tags.
...
+* Updated tlddoc generation to support jakarta.
 * General specification document cleanup.
-* Clean up issues found during the Jakarta EE9 review ballot process
-* Update tagdocs to include EFSL
-* Update tlddoc generation to jakarta.
-* Remove any deprecated methods from the API if they exist.
-* Rename java.sun.com / xmlns.jcp.org taglib URIs to new jakarta.tags.\* URNs. [Issue 144](https://github.com/eclipse-ee4j/jstl-api/issues/144)
-    * This alone should require a major release.
-* Other small clean up issues
+
```
[the remark this answers is not in the document]
Comment (arjantijms) [reply tags/3.0/_index.md:30]: For some other specs I added the GitHub issues where appropriate:

*  https://github.com/jakartaee/specifications/pull/460/files#diff-031c5337b39b2a8d742e677df77f3a099cb906519800f4800b72b30af8e782bd
* https://github.com/jakartaee/specifications/pull/461/files#diff-c93f4dcb13475d9a13f9dc8e92dbd2e0b289f20dc9d06d7c319c3d544b0cc7c5
* https://github.com/jakartaee/specifications/pull/459/files#diff-442cedce2cdf5cc01aea2bcfe1dc928904c68c0477da2dea9f07f7d3d2c991b2

Might be good for consistency (although not required of course) to have somewhat of the similar format?

[the remark this answers is not in the document]
Comment (volosied) [reply tags/3.0/_index.md]: I added links to the changes labeled under `Changed between Jakarta Standard Tag Library 2.0
and Jakarta Standard Tag Library 3.0` 

As for the other changes, there were multiple aspects to it, so I didn't bother adding multiple links.

**Comment by Emily-Jiang** [comment]
* On ballot completion, the specification committee mentor:
 - [ ] adds this final checklist to the main PR.
 - [ ] adds the `approved` label to the PRs, and sends out the Ballot Summary per this [template](#ballot-summary-email-template) to the [public Jakarta EE Specification Committee email list](jakarta.ee-spec@eclipse.org)
 - [ ] calculates the staged EFTL TCK signature and promotes it to the committee download area
  using the https://ci.eclipse.org/jakartaee-spec-committee/job/promote-release/ job. Manually editing the jenkins Build Information will help identify the build (ie. Mail 2.0 or CDI 3.0).
 - [ ] merges the specification (and apidocs) PRs, ensuring the "date:" field in the _index.md file has an appropriate value to allow publishing.
 - [ ] updates the specification page with the ballot results. This is normally done via a separate PR that should be reviewed, approved, and merged.
 - [ ] notifies the EMO of the ballot results by email to [emo@eclipse-foundation.org](emo@eclipse-foundation.org). Just forward the ballot summary note sent earlier to the public Spec Committee email list.
 - [ ] creates an issue in the specification project that includes the following checklist for the specification project team:
 \<include url of Issue here>

**Pull request merged**